lib/IRGen/IRGenSIL.cpp

```diff
@@ -3733,6 +3733,16 @@ void IRGenSILFunction::visitDebugValueAddrInst(DebugValueAddrInst *i) {
   SILType SILTy = SILVal->getType();
   auto RealType = SILTy.getASTType();
 
+  if (IGM.DebugInfo &&
+      IGM.getOptions().DebugInfoFormat == IRGenDebugInfoFormat::CodeView)
+    // In LLDB, inout types are emitted as their dereferenced type. However, it
+    // is better for CodeView if inout types are emitted as a reference to the
+    // appropriate type, which is accomplished when RealType is TypeKind::InOut.
+    // FIXME: This should only happen when ``i`` is an inout variable. Is there
+    //        another way that ``visitDebugValueAddrInst`` could be called on
+    //        ``i``? If so, is there a way to check if it is an inout variable?
+    RealType = CanInOutType::get(RealType);
```

Comment:
This is really weird. Do you have a full example?

Reply:
I was just wondering if there was an example where `visitDebugValueAddrInst` is called on an instruction that does not have an `inout` type. Because my code will convert `RealType` to an `inout` type.

Reply:
This is not the right location to have any CodeView-specific code. Frankly I'm a bit confused by this: A debug_value_addr instruction doesn't necessarily mean that that the variable has an inout type. it sounds like you might want to modify `DebugTypeInfo::isImplicitlyIndirect()`? But even that looks sketchy. Why do you need to treat this differently in CodeView at all?

Reply:
I was having a problem because I was hitting the `case TypeKind:InOut` for the types of a functions arguments, but not for the local variable of the same type. The `inout` local variable was actually emitted as its desugared type, which is incorrect.